### PR TITLE
Downloader feature

### PR DESCRIPTION
**Pull request by TheLustriVA**
# Built-in Downloader

## Downloader

I was writing a download script anywhere so I decided to flesh it out and contribute it back. 

Usage for `download.py` script is described in the `README.md` and `index.md`. The script also outputs to `manifest.txt`.

## Dependency control

I was using Pipenv for dependency control so I added the dependencies for the rest of the scripts while I was there. If you haven't used Pyenv and Pipenv before, I thoroughly recommend you give it a try. It simplifies controlling virtual environments enormously.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,7 +175,7 @@ urlretrieve(part_url, f'part-{part_id:06}.zip')
 shutil.unpack_archive(f'part-{part_id:06}.zip', f'part-{part_id:06}')
 ```
 
-### Method 3. Use `metadata.parquet` (Text Only)
+### Method 4. Use `metadata.parquet` (Text Only)
```
Comment (xiaohk) [inline README.md:178]: Here should still be Method 3 :)

File: download.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
+"""_summary_
```
Comment (xiaohk) [inline download.py:1]: Suggest adding your name/github username here as author, and adding MIT license.

File: download.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+
+# This is a common pattern in Python. It allows you to run the main function of your script by
+# running the script through the interpreter. It also allows you to import the script into the
+# interpreter
+# without automatically running the main function.
+if __name__ == "__main__":
+    main(index, range_max, output, unzip)
+
+"""
```
Comment (xiaohk) [inline download.py:176]: Thanks for acknowledgements! Suggest removing it from this file though 😄

File: index.md [hunk 4]
```diff
@@ -198,7 +266,7 @@ urlretrieve(part_url, f'part-{part_id:06}.zip')
 shutil.unpack_archive(f'part-{part_id:06}.zip', f'part-{part_id:06}')
 ```
 
-#### Method 3. Use `metadata.parquet` (Text Only)
+#### Method 4. Use `metadata.parquet` (Text Only)
```
Comment (xiaohk) [inline index.md:269]: Still Method 3

**Review by xiaohk — COMMENTED** [review]
Wow, thank you so much @TheLustriVA!! It looks very nice, and I am sure the script will be super helpful for others. I added some comments. Can you address them and add a PR directly to the main branch?

File: README.md (hunk 1 above)
Comment (TheLustriVA) [reply README.md:178]: Will change :D

File: download.py (hunk 2 above)
Comment (TheLustriVA) [reply download.py:1]: Good thinking.

File: download.py (hunk 3 above)
Comment (TheLustriVA) [reply download.py:176]: Can do :D

File: index.md (hunk 4 above)
Comment (TheLustriVA) [reply index.md:269]: Gotcha.

File: README.md (hunk 1 above)
Comment (TheLustriVA) [reply README.md:178]: So that it still makes sense, I've removed the original Method 2 for manually downloading the data,

**Comment by TheLustriVA** [comment]
Those changes have been made and a PR made to the main branch ready for review :D

**Comment by TheLustriVA** [comment]
@xiaohk Sorry mate, I don't have permission to do a pull request into the main branch. You'd need to merge this one after accepting the PR, sorry.

**Comment by xiaohk** [comment]
> @xiaohk Sorry mate, I don't have permission to do a pull request into the main branch. You'd need to merge this one after accepting the PR, sorry.

Hmm weird, I think you should have permission. Okay, can you add your 6d0a01bbe137f6db7fe47677349070c296d5477a commit to the `downloader-feature` branch? Thanks!

**Pull request closed without merging**